### PR TITLE
Fix custom primary key

### PR DESCRIPTION
**Pull request by CaroFG**

### CODE REVIEW

File: lib/meilisearch-rails.rb [hunk 1]
```diff
@@ -293,9 +293,10 @@ def additional_indexes
   # this class wraps an Algolia::Index object ensuring all raised exceptions
   # are correctly logged or thrown depending on the `raise_on_failure` option
   class SafeIndex
-    def initialize(index_uid, raise_on_failure)
+    def initialize(index_uid, raise_on_failure, options)
       client = MeiliSearch.client
-      @index = client.get_or_create_index(index_uid)
+      options[:primary_key].nil? ?  primary_key = {} : primary_key = { primaryKey: options[:primary_key].to_s }
+      @index = client.get_or_create_index(index_uid, primary_key)
```
Comment (curquiza) [inline lib/meilisearch-rails.rb:299]:
```suggestion
      primary_key = options[:primary_key].to_s ||  MeiliSearch::IndexSettings::DEFAULT_PRIMARY_KEY
      @index = client.get_or_create_index(index_uid, { primaryKey: primary_key })
```

File: lib/meilisearch-rails.rb [hunk 2]
```diff
@@ -895,6 +897,11 @@ def ms_primary_key_changed?(o, options = nil)
       changed.nil? ? false : changed
     end
 
+    def ms_pk(options = nil) 
+      options[:primary_key].nil? ? pk = 'id' : pk = options[:primary_key] 
+      pk
```
Comment (curquiza) [inline lib/meilisearch-rails.rb:902]:
```suggestion
      options[:primary_key] || MeiliSearch::IndexSettings::DEFAULT_PRIMARY_KEY
```

File: lib/meilisearch-rails.rb (hunk 1 above)
Comment (curquiza) [reply lib/meilisearch-rails.rb:299]: You should add `DEFAULT_PRIMARY_KEY` in `MeiliSearch::IndexSettings` of course 😇

**Review by curquiza — CHANGES_REQUESTED** [review]

**Review by curquiza — APPROVED** [review]
Well done Caro :)

**Comment by CaroFG** [comment]
bors merge

**Pull request merged**